ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Byron/Forge.hs

```diff
@@ -93,6 +94,31 @@ forgeGenesisEBB (WithEBBNodeConfig cfg) curSlot =
           . CC.Slot.fromSlotNumber pbftEpochSlots
           $ coerce curSlot
 
+-- | Internal helper data type for 'forgeBlock' used to accumulate the
+-- different kinds of block payloads that can be found in a given collection
+-- of Byron 'GenTx's.
+--
+-- n.b. This data type is not to be exposed from this module.
+data BlockPayloads = BlockPayloads
+  { bpTxs         :: ![CC.UTxO.TxAux]
+  , bpDlgCerts    :: ![CC.Delegation.Certificate]
+  , bpUpVotes     :: ![CC.Update.Vote]
+  , bpUpProposal  :: !(Maybe CC.Update.Proposal)
+    -- ^ 'Just' if there is at least one 'CC.Update.Proposal' in a list of
+    -- Byron 'GenTx's and 'Nothing' if there are none. It is worth noting that
+    -- if we encounter multiple 'CC.Update.Proposal's in a collection of
+    -- 'GenTx's, this value will be that of the last 'CC.Update.Proposal'
+    -- encountered.
```

Comment:
Perhaps `forgeBlock` should actually return an error in the event that there are multiple `CC.Update.Proposal`s provided in a list of Byron `GenTx`s since this should _never_ happen. Or maybe it's okay because this will fail the validation phase. I'm not sure what level of validation we'd like to perform here.

Reply:
Throwing an error is the way to go.

Reply:
See my comment about the TODO